src/client/src/tscSQLParser.c

```diff
@@ -4061,6 +4057,12 @@ int32_t getTableIndexByName(SStrToken* pToken, SQueryInfo* pQueryInfo, SColumnIn
 }
 
 int32_t getColumnIndexByName(const SStrToken* pToken, SQueryInfo* pQueryInfo, SColumnIndex* pIndex, char* msg) {
```

Comment:
Check the validity of pToken within getColumnIndexByName here, which rejects tokens other than strings.